src/Sentry.Maui.CommunityToolkitMvvm/CtMvvmMauiElementEventBinder.cs

```diff
@@ -0,0 +1,151 @@
+                TryBindTo(refresh.Command, bind);
+                break;
+
+            case SearchBar searchBar:
+                TryBindTo(searchBar.SearchCommand, bind);
+                break;
+
+            default:
+                TryGestureBinding(element, bind);
+                break;
+        }
+    }
+
+    private static void TryGestureBinding(VisualElement element, bool bind)
+    {
+        if (element is IGestureRecognizers gestureRecognizers)
+        {
+            foreach (var gestureRecognizer in gestureRecognizers.GestureRecognizers)
+            {
+                TryBindTo(gestureRecognizer, bind);
+            }
+        }
+    }
+
+    private static void TryBindTo(IGestureRecognizer recognizer, bool bind)
+    {
+        switch (recognizer)
+        {
+            case TapGestureRecognizer tap:
+                TryBindTo(tap.Command, bind);
+                break;
+
+            case SwipeGestureRecognizer swipe:
+                TryBindTo(swipe.Command, bind);
+                break;
+
+            // no commands
+            //case PinchGestureRecognizer pinch
+            //case PanGestureRecognizer pan
+            case DragGestureRecognizer drag:
+                TryBindTo(drag.DragStartingCommand, bind); // unlikely to ever be async
+                TryBindTo(drag.DropCompletedCommand, bind);
+                break;
+
+            case PointerGestureRecognizer pointer:
+                TryBindTo(pointer.PointerPressedCommand, bind);
+                TryBindTo(pointer.PointerReleasedCommand, bind);
+                TryBindTo(pointer.PointerEnteredCommand, bind); // unlikely to ever be async
+                TryBindTo(pointer.PointerExitedCommand, bind);
+                break;
+        }
+    }
+
+    private static List<IAsyncRelayCommand> _refs = [];
+    private static void TryBindTo(ICommand? command, bool bind)
+    {
+        if (command is IAsyncRelayCommand relayCommand)
+        {
+            if (bind)
+            {
+                // necessary for collectionview buttons
+                if (!_refs.Contains(relayCommand))
+                {
+                    _refs.Add(relayCommand);
+                    relayCommand.PropertyChanged += RelayCommandOnPropertyChanged;
+                }
+            }
+            else
+            {
+                _refs.Remove(relayCommand);
+                relayCommand.PropertyChanged -= RelayCommandOnPropertyChanged;
+            }
+        }
+    }
+
+    private static ConcurrentDictionary<IAsyncRelayCommand, (ITransactionTracer Transaction, ISpan Span)> _contexts = new();
+
+    private static void RelayCommandOnPropertyChanged(object? sender, PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(IAsyncRelayCommand.IsRunning))
+        {
+            var relay = (IAsyncRelayCommand)sender!;
+
+            if (relay.IsRunning)
+            {
+                var transaction = SentrySdk.StartTransaction("ctmvvm", "asynccommand");
```

Comment:
for testability we usually take `IHub` in an internal constructor, passing `HubAdapter.Instance` from the public one not to have to expose this to the caller.

```suggestion
                var transaction = _hub_.StartTransaction("ctmvvm", "asynccommand");
```

Reply:
I agree that normally that is a good path.  What I do worry about is allocating more pointers per control.  Using the static events keeps our footprint lower in large pages with many controls.  This probably isn't going to be a huge issue, but I wanted to start on the happy path first.  Automated testing with this thing is difficult at best.  Happy to change this, but let me know your thoughts.

Reply:
I've added the instance based methods and ihub